central/scannerdefinitions/handler/handler.go

```diff
@@ -654,3 +664,39 @@ func getOfflineFileVersion(mf *os.File) (string, error) {
 	}
 	return m.Version, nil
 }
+
+func validateV4DefsVersion(zipPath string) error {
+	zipR, err := zip.OpenReader(zipPath)
+	if err != nil {
+		return errors.Wrap(err, "couldn't open file as zip")
+	}
+	defer utils.IgnoreError(zipR.Close)
+
+	for _, zipF := range zipR.File {
+		if strings.HasPrefix(zipF.Name, scannerV4DefsPrefix) {
+			defs, _, err := openFromArchive(zipPath, zipF.Name)
+			if err != nil {
+				return errors.Wrap(err, "couldn't open v4 offline defs manifest.json")
+			}
+			utils.IgnoreError(defs.Close)
+			mf, removeDefs, err := openFromArchive(defs.Name(), "manifest.json")
+			if err != nil {
+				return errors.Wrap(err, "couldn't open v4 offline defs manifest.json")
+			}
+			offlineV, err := getOfflineFileVersion(mf)
+			utils.IgnoreError(mf.Close)
+			removeDefs()
+			if err != nil {
+				return errors.Wrap(err, "couldn't get v4 offline defs version")
+			}
+			v := minorVersionPattern.FindString(version.GetMainVersion())
+			if offlineV != "dev" && offlineV != v {
+				msg := fmt.Sprintf("failed to upload offline file bundle, uploaded file is version: %s and system version is: %s; "+
+					"please upload an offline bundle version: %s, consider using command roxctl scanner download-db", offlineV, version.GetMainVersion(), v)
+				log.Errorf(msg)
+				return errors.New(msg)
+			}
```

Comment:
For future consideration (nice to have):

`offlineV` may have a version similar to `4.3.x-1354-gc7c197b59e` if generated via a PR label, since we're extracting `X.Y` from `version.GetMainVersion()` may make sense to do the same for `offlineV`.  Or perhaps even making use of the `version.Variants()` of the main version and comparing each variant to the `offlineV`. 

However, this assumption may not hold true in the future given the bundle versioning strategy is unclear / undefined currently.  We may end up introducing logic to use the 'closest' version of a bundle vs. an exact version match like we're doing today.